pkg/services/ngalert/accesscontrol/rules_test.go

```diff
@@ -325,29 +420,18 @@ func TestAuthorizeRuleChanges(t *testing.T) {
 				permissionCombinations = permissionCombinations[0 : len(permissionCombinations)-1] // exclude all permissions
 				for _, missing := range permissionCombinations {
 					ac := &recordingAccessControlFake{}
-					srv := RuleService{
-						genericService{ac: ac},
-					}
+					srv := NewRuleService(ac)
 					err := srv.AuthorizeRuleChanges(context.Background(), createUserWithPermissions(missing), groupChanges)
 
 					assert.Errorf(t, err, "expected error because less permissions than expected were provided. Provided: %v; Expected: %v; Diff: %v", missing, permissions, cmp.Diff(permissions, missing))
 					require.NotEmptyf(t, ac.EvaluateRecordings, "Access control was supposed to be called but it was not")
 				}
 			})
 
-			ac := &recordingAccessControlFake{
```

Comment:
This idea doesn't work anymore since not all calls to evaluate are meant to success when checking receiver Auth. It checks `preconditions` and `all` separately. Instead replaced with a real ac.